src/views/SignIn/index.tsx

```diff
@@ -0,0 +1,128 @@
+import { useEffect, useState } from "react";
+import { Link, Route, Routes, useLocation, useNavigate } from "react-router-dom";
+
+import { Button } from "../../components";
+import { AuthResult, useAuth, useToasts } from "../../providers";
+import { ForgottenPassword } from "./ForgottenPassword";
+
+export const SignIn: React.FC = () => {
+  const { toast, untoast } = useToasts();
+  const { signIn, isAuthenticated } = useAuth();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  // note: perform the check only on mounting (and **not** on subsequent changes) to avoid race conditions with the router
+  useEffect(() => {
+    if (isAuthenticated) navigate("/browse");
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  return (
+    <div className="mx-auto">
+      <Routes>
+        <Route
+          path="forgotten"
+          element={
+            <ForgottenPassword
+              prefill={() => {
+                setEmail("jane@petite.com");
+                setPassword("supersafe");
+              }}
+            />
+          }
+        />
+      </Routes>
+
+      <form
+        className="flex flex-col items-center max-w-sm"
+        onSubmit={async (e) => {
+          e.preventDefault();
```

Comment:
I would move it to a separate function or maybe to a hook, for example smth like `useSignup()` ?

Reply:
Thanks, somehow forgot to wrap this in a `useCallback`.